blog/articles/orchestrating-cloud-events-with.md

```diff
@@ -0,0 +1,194 @@
+
+![Websockets push notification](https://github.com/salaboy/orchestrating-cloud-events/blob/master/imgs/tickets-push-notification-websocket.png?raw=true)
+
+The above image shows the notification on the frontend, which was generated by the workflow engine, routed to the website backend and forwarded using WebSockets to the website frontend in the customer browser.
+
+The top one, with solid lines, is an interrupting event, meaning that it will cause the normal flow of the model to be discarded and continue only to the steps defined from the timer, in this case, **Reservation Time Out**. For this scenario, the interrupting timer is set to be triggered after X amount (for the presentation this was set to 2 minutes) of time after the reservation of the tickets was made. After the reservation time out the website redirects the customer back to the beginning.
+
+For both timers, you get full traceability (e.g. when and how oftenthey were triggered). You can see these details graphically in **Operate**:
+
+![Workflow model v2 in Operate](https://github.com/salaboy/orchestrating-cloud-events/blob/master/imgs/tickets-v2-timers-operate.png?raw=true)
+
+You can see that the non-interrupting timer was triggered several times, generating payment reminders in the frontend. Even if we don’t implement the frontend notifications, you can use these timers to measure and classify how much time it actually takes your customer to make the payment. This can be really useful to classify different categories, for example, how much time it takes, on average, for people to provide payment details.
+
+A big difference between these timers and [Cronjobs (for example in Kubernetes)](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/) is that these timers are contextual to the state that they are associated with, meaning that as soon as the **Payment Sent** CloudEvent is received, both timers are automatically cancelled and garbage collected. For this particular scenario, you want notifications about the payment to be sent but also make sure the reservation doesn’t time out if the payment was sent. The workflow engine deals with these very common requirements in a transparent way for the user who is interested in registering time-based events as part of their applications.
+
+## Orchestrate
+
+The **Payment Reminder** and **Reservation Time Out** states, shown in the model above, represent when we cross the line from simply listening to events to actually emitting CloudEvents from a workflow model. Now the workflow model is not only helping you to visualize and understand what your Cloud Native applications are doing but driving and interacting with your services.
```

Comment:
```suggestion
The **Payment Reminder** and **Reservation Time Out** states, shown in the model below, represent when we cross the line from simply listening to events to actually emitting CloudEvents from a workflow model. Now the workflow model is not only helping you to visualize and understand what your Cloud Native applications are doing but driving and interacting with your services.
```

Reply:
adding as a separate revision because I am not sure if you want to commit this change. I noticed you can see these status in both the model above, and the one right below this paragraph, so up to you.

Reply:
I am ok with the one above